extensions/obsidian/src/createnote.tsx

```diff
@@ -0,0 +1,186 @@
+import {
+  List,
+  ActionPanel,
+  getPreferenceValues,
+  PushAction,
+  Form,
+  SubmitFormAction,
+  useNavigation,
+  showToast,
+  ToastStyle,
+  AlertOptions,
+  confirmAlert,
+  Icon,
+} from "@raycast/api";
+import fs from "fs";
+import path from "path";
+
+interface Vault {
+  name: string;
+  key: string;
+  path: string;
+}
+
+interface Preferences {
+  vaultPath: string;
+  prefPath: string;
+  prefTag: string;
+  tags: string;
+  primaryAction: string;
+}
+
+interface FormValue {
+  path: string;
+  name: string;
+  content: string;
+  tags: Array<string>;
+}
+
+function getVaultNameFromPath(vaultPath: string) {
+  return vaultPath.substring(vaultPath.lastIndexOf(path.sep) + 1);
```

Comment:
```suggestion
  return vaultPath.split(path.sep).filter(function(i){return i}).pop();
```
If `vaultPath` ends with / this will return "", since we can't control what the user input, then we should take care of that if possible